vault/generate_root.go

```diff
@@ -221,6 +227,9 @@ func (c *Core) GenerateRootUpdate(ctx context.Context, key []byte, nonce string,
 	if c.Sealed() && !c.recoveryMode {
 		return nil, consts.ErrSealed
 	}
+	if !c.Sealed() && c.recoveryMode {
```

Comment:
Same here. We may have to use `!c.barrier.Sealed()`

Reply:
Not following, what's the same here?

Reply:
In recovery mode, we only care about barrier being unsealed. Core will likely remain sealed throughout.

Reply:
Ah, didn't see your other comment in that thread.  Yeah, I was setting `c.sealed=0` but I like your solution better.